SS14.ServerHub/Controllers/ServerListController.cs

```diff
@@ -124,6 +124,24 @@ parsedAddress.Scheme is not (Ss14UriHelper.SchemeSs14 or Ss14UriHelper.SchemeSs1
         var addressEntity =
             await _dbContext.AdvertisedServer.SingleOrDefaultAsync(a => a.Address == advertise.Address);
 
+        if (senderIp != null)
+        {
+            // Check the current number of advertised servers from this IP.
+            var count = await _dbContext.AdvertisedServer
+                .Where(s => s.AdvertiserAddress == senderIp)
+                .Where(s => s.Expires > DateTime.UtcNow)
+                .CountAsync();
+
+            var isServerRenewingAdvertisement = addressEntity != null && addressEntity.Expires > DateTime.UtcNow;
+
+            if (!isServerRenewingAdvertisement
+                && count >= HubOptions.MaxServersPerIp
+                && !await CheckExemptFromMaxAdvertisements(parsedAddress))
+            {
+                return Unauthorized($"You cannot advertise more then {HubOptions.MaxServersPerIp} servers from one IP address, please contact us if you require an increase.");
```

Comment:
1-letter grammar nitpick
```suggestion
                return Unauthorized($"You cannot advertise more than {HubOptions.MaxServersPerIp} servers from one IP address, please contact us if you require an increase.");
```